Assets/BossRoom/Scripts/Shared/Net/UnityRelayUtilities.cs

```diff
@@ -0,0 +1,82 @@
+﻿using System;
+using System.Threading.Tasks;
+using UnityEngine;
+using Unity.Services.Relay;
+using Unity.Services.Relay.Allocations;
+using Unity.Services.Relay.Models;
+
+namespace BossRoom
+{
+    public static class RelayJoinCodeThing
+    {
+        public static string RelayJoinCode = String.Empty;
+    }
+    public class RelayUtility
+    {
+        async public static
+            Task<(string ipv4address, ushort port, byte[] allocationIdBytes, byte[] connectionData, byte[] key, string
+                joinCode)> AllocateRelayServerAndGetJoinCode(int maxConnections, string region = null)
+        {
+            Response<AllocateResponseBody> allocationResponse;
+            Response<JoinCodeResponseBody> createJoinCodeResponse;
+            try
+            {
+                allocationResponse =
+                    await RelayService.AllocationsApiClient.CreateAllocationAsync(
+                        new CreateAllocationRequest(new AllocationRequest(maxConnections)));
+            }
+            catch
+            {
+                Debug.LogError("Relay create allocation request failed");
+                throw;
+            }
+
+            var allocation = allocationResponse.Result.Data.Allocation;
+
+            Debug.Log($"server: {allocation.ConnectionData[0]} {allocation.ConnectionData[1]}");
+            Debug.Log($"server: {allocation.AllocationId}");
+
+            try
+            {
+                createJoinCodeResponse = await RelayService.AllocationsApiClient.CreateJoincodeAsync(
+                    new CreateJoincodeRequest(
```

Comment:
Who should we tell there's a typo in their API? It's missing a capital C for Code

Reply:
They auto generate this API so its a tool being used... so that is how the tool generated the code for this API. So no spelling error from the point of their tool.

Reply:
This API has been updated to use a newer API that Relay has pushed.

Reply:
much cleaner, thank you